metadata-ingestion/src/datahub/ingestion/source_config/sql/snowflake.py

```diff
@@ -75,20 +79,48 @@ class BaseSnowflakeConfig(BaseTimeWindowConfig):
         default=None, exclude=True
     )
     authentication_type: str = "DEFAULT_AUTHENTICATOR"
-    host_port: str
+    host_port: Optional[str] = None  # Deprecated
+    account_id: Optional[
+        str
+    ] = None  # Once host_port is removed this will be made mandatory
     warehouse: Optional[str]
     role: Optional[str]
     include_table_lineage: Optional[bool] = True
     include_view_lineage: Optional[bool] = True
     connect_args: Optional[Dict] = pydantic.Field(default=None, exclude=True)
 
-    @pydantic.validator("host_port", always=True)
-    def host_port_is_valid(cls, v, values, **kwargs):
+    def get_account(self) -> str:
+        if self.account_id is not None:
+            return self.account_id
+        # This is handled in root validator but needed to do this due to linting
+        raise ConfigurationError("account_id missing")
+
+    def _clean_account(cls, v: str) -> str:
         v = remove_protocol(v)
         v = remove_trailing_slashes(v)
         v = remove_suffix(v, ".snowflakecomputing.com")
         return v
 
+    @pydantic.root_validator
+    def one_of_host_port_or_account_id_is_required(cls, values):
+        host_port = values.get("host_port")
+        if host_port is not None:
+            logger.warning(
```

Comment:
Is this case being covered in the tests?

Reply:
added